crates/scmi/src/main.rs

```diff
@@ -70,18 +71,15 @@ struct ScmiArgs {
 }
 
 pub struct VuScmiConfig {
-    socket_path: String,
+    socket_path: PathBuf,
     devices: DeviceDescription,
 }
 
 impl TryFrom<ScmiArgs> for VuScmiConfig {
     type Error = String;
 
     fn try_from(cmd_args: ScmiArgs) -> Result<Self> {
-        if cmd_args.socket_path.is_none() {
```

Comment:
Since this change, a scmi test is failing. I don't think this makes sense as just setting the argument as not optional would make clap require it i believe; cc @mz-pdm

Reply:
This is because of `help_devices` option, which is examined before calling this. So no, you cannot remove this check now. We discussed previously that it would be nice to integrate the device help with clap completely but the pending question is how to do it, without crippling the help text provided to the user.